test/subscription_map.hpp

```diff
@@ -439,31 +439,39 @@ class multiple_subscription_map
     // Insert a key => value at the specified subscription path
     // returns the handle and true if key was inserted, false if key was updated
     template <typename V>
-    std::pair<handle, bool> insert_or_update(MQTT_NS::string_view subscription, Key const &key, V&& value) {
+    std::pair<handle, bool> insert_or_update(MQTT_NS::string_view subscription, Key&& key, V&& value) {
         auto path = this->find_subscription(subscription);
         if (path.empty()) {
             auto new_subscription_path = this->create_subscription(subscription);
-            new_subscription_path.back()->second.value[key] = std::forward<V>(value);
+            new_subscription_path.back()->second.value.insert(std::make_pair(std::forward<Key>(key), std::forward<V>(value)));
             ++this->map_size;
             return std::make_pair(this->path_to_handle(new_subscription_path), true);
```

Comment:
I do not understand where you suggest the move, new_subscription_path is not send directly to pair, path_to_handle takes const &

Reply:
```
return std::make_pair(this->path_to_handle(new_subscription_path), true);
```

Should be

```
return std::make_pair(this->path_to_handle(std::move(new_subscription_path)), true);
```

Even if `path_to_handle` currently takes a `const&`, it may not always, so using `std::move()` is optimal.